lib/streamlit/error_util.py

```diff
@@ -35,20 +35,83 @@
 _GENERIC_UNCAUGHT_EXCEPTION_TEXT = "This app has encountered an error. The original error message is redacted to prevent data leaks.  Full error details have been recorded in the logs (if you're on Streamlit Cloud, click on 'Manage app' in the lower right of your app)."
 
 
+def _print_rich_exception(e: BaseException):
+    from rich import panel, box  # type: ignore
+
+    # Monkey patch the panel to use our custom box style
+    class ConfigurablePanel(panel.Panel):
+        def __init__(
+            self,
+            renderable,
+            box=box.Box("────\n    \n────\n    \n────\n────\n    \n────\n"),
+            **kwargs,
+        ):
+            super(ConfigurablePanel, self).__init__(renderable, box, **kwargs)
+
+    from rich import traceback as rich_traceback
+
+    rich_traceback.Panel = ConfigurablePanel
+
+    # Configure console
+    from rich.console import Console  # type: ignore
+
+    console = Console(
+        color_system="256",
+        force_terminal=True,
+        width=88,
+        no_color=False,
+        tab_size=8,
+    )
+
+    from streamlit import script_runner
+
+    # Print exception via rich
+    console.print(
+        rich_traceback.Traceback.from_exception(
+            type(e),
+            e,
+            e.__traceback__,
+            width=88,
+            show_locals=False,
+            max_frames=100,
+            word_wrap=False,
+            extra_lines=3,
+            suppress=[script_runner],  # Ignore script runner
+        )
+    )
+
+
 def handle_uncaught_app_exception(e: BaseException) -> None:
     """Handle an exception that originated from a user app.
     By default, we show exceptions directly in the browser. However,
     if the user has disabled client error details, we display a generic
     warning in the frontend instead.
     """
+
+    errorLogged = False
+
+    if config.get_option("logger.enableRich"):
+        try:
+            # Print exception via rich
+            # Rich is only a soft dependency
+            # -> if not installed, we will use the default traceback formatting
+            _print_rich_exception(e)
+            errorLogged = True
+        except Exception:
```

Comment:
nit: can the exact `Exception` type we catch here be scoped down a bit (to a `ModuleNotFoundError` and other expected error types) so that we don't accidentally swallow other errors?

Reply:
Catching all Exceptions here is intentional (I will add a comment). This is using some lower-level APIs of rich and it is hard to guarantee that it works fine with all rich versions. The impact of breaking here would be much bigger than showing the default formatted stack trace. So, I definitely don't want to leave any possibility of breaking here.